java/src/jmri/managers/AbstractStringIOManager.java

```diff
@@ -32,6 +34,67 @@ public int getXMLOrder() {
         return Manager.STRINGIOS;
     }
 
+    /** {@inheritDoc} */
+    @Override
+    @Nonnull
+    public StringIO provideStringIO(@Nonnull String name) {
+        var t = getStringIO(name);
+        if (t == null) {
+            t = newStringIO(makeSystemName(name), null);
+        }
+        return t;
+    }
+
+    private StringIO getStringIO(@Nonnull String name) {
+        var t = getByUserName(name);
+        if (t != null) {
+            return t;
+        }
+        return getBySystemName(name);
+    }
+
+    /**
+     * Create a New StringIO.
+     * {@inheritDoc}
+     */
+    @Override
+    @Nonnull
+    final public StringIO newStringIO(@Nonnull String systemName, String userName) throws IllegalArgumentException {
+        log.debug(" newStringIO(\"{}\", \"{}\")", systemName, (userName == null ? "null" : userName));
+        Objects.requireNonNull(systemName, "SystemName cannot be null. UserName was "
+                + (userName == null ? "null" : userName));  // NOI18N
+        systemName = validateSystemNameFormat(systemName);
+        // return existing if there is one
+        StringIO s;
+        if (userName != null) {
+            s = getByUserName(userName);
+            if (s != null) {
+                if (getBySystemName(systemName) != s) {
+                    log.error("inconsistent user ({}) and system name ({}) results; userName related to ({})", userName, systemName, s.getSystemName());
+                }
+                return s;
+            }
+        }
+        s = getBySystemName(systemName);
+        if (s != null) {
+            if ((s.getUserName() == null) && (userName != null)) {
+                s.setUserName(userName);
+            } else if (userName != null) {
+                log.warn("Found StringIO via system name ({}) with non-null user name ({}). StringIO \"{}({})\" cannot be used.",
+                        systemName, s.getUserName(), systemName, userName);
```

Comment:
I suggest throwing an `IllegalArgumentException` here.

Reply:
This is the counterpart of the suggestion above. This is probably a good idea, but this is what all of these managers do now: They return the existing object with the requested name, and issue that error.  We can change that behavior here, but I'm not sure I want this manager to behave like every other one in the system.  And changing all those other ones may well break scripts, etc where the user hasn't noticed any problems.  Are we sure we want to change just this one?

And thank you for the careful reviews!

Reply:
I thought that it would be good to change all the managers but I realize that it could cause problems with scripts.
I'm fine with keeping it as is.